src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.vb

```diff
@@ -0,0 +1,435 @@
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub MultiDimensionalArrayCreation_WithInitializer()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F()
+        Dim b As Byte(,,) = New Byte(,,) {{{1, 2, 3}}, {{4, 5, 6}}}'BIND:"New Byte(,,) {{{1, 2, 3}}, {{4, 5, 6}}}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 3, Element Type: System.Byte) (OperationKind.ArrayCreationExpression, Type: System.Byte(,,))
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 3)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+    IArrayInitializer (OperationKind.ArrayInitializer)
+      IArrayInitializer (OperationKind.ArrayInitializer)
+        IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Byte, Constant: 1)
+          ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+        IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Byte, Constant: 2)
+          ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+        IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Byte, Constant: 3)
+          ILiteralExpression (Text: 3) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 3)
+    IArrayInitializer (OperationKind.ArrayInitializer)
+      IArrayInitializer (OperationKind.ArrayInitializer)
+        IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Byte, Constant: 4)
+          ILiteralExpression (Text: 4) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 4)
+        IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Byte, Constant: 5)
+          ILiteralExpression (Text: 5) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 5)
+        IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Byte, Constant: 6)
+          ILiteralExpression (Text: 6) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 6)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub ArrayCreationOfSingleDimensionalArrays()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F()
+        Dim a = {{1, 2, 3}, {4, 5, 6}}'BIND:"{{1, 2, 3}, {4, 5, 6}}"
```

Comment:
>{{1, 2, 3}, {4, 5, 6}} [](start = 16, length = 22)

Do we have tests for erroneous array literals (mismatched element types, mismatched dimensions, etc.)?